### ISSUE TITLE
use_field_init_shorthand is listed as default: false but is enabled by default.

### ISSUE THREAD

**Issue by rukai** [rust-lang/rustfmt#4948]
When I run `cargo fmt -- --help=config` it shows use_field_init_shorthand as `default: false`.
However running `cargo fmt` will format as if use_field_init_shorthand is true.

I think use_field_init_shorthand should list the default as true to reflect reality.

**Comment by calebcartwright** [comment]
> However running cargo fmt will format as if use_field_init_shorthand is true.

This does not sound accurate. Can you please provide a minimal example that reproduces the behavior you're describing?

**Comment by rukai** [comment]
Ah, sorry, the issue is not exactly as I described, but it is still an issue regardless.

It seems `use_field_init_shorthand: false` functions as an `off` despite the  documentation clearly describing it as forcing longhand formatting. https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#use_field_init_shorthand

consider:
```rust
struct Foo {
    a: i32,
}

fn main() {
    let foo = Foo { a };
}
```

and

```rust
struct Foo {
    a: i32,
}

fn main() {
    let foo = Foo { a: a };
}
```

which both remain unchanged after running `cargo fmt`

**Comment by calebcartwright** [comment]
> both remain unchanged after running cargo fmt

Can you please clarify what configuration options you are using, and what your expected behavior is?

It seems like in the first snippet you are expecting `use_field_init_shorthand` to _convert_ shorthand to longhand which is not what the option does. The example snippets shown in the config docs only highlight formatting style that will remain as-is/pass `--check` with the corresponding option value. They do _not_ imply any reformatting except in cases where the documentation contains a section/comment/etc. that highlights a "before/pre-formatting".

In the second snippet, I'd imagine you would expect that to be converted to:

```rust
fn main() {
    let foo = Foo { a };
}
```

which is precisely what the option is supposed to do, and what it will do so long as you have `use_field_init_shorthand` overridden with the non-default value of `true`

**Comment by rukai** [comment]
I'm not saying to use this exact wording but the documentation would make a lot more sense if we wrote it something like this:
![image](https://user-images.githubusercontent.com/5120858/132435271-b524ab4b-4577-4cbc-80b0-4ddf46548310.png)

**Comment by calebcartwright** [comment]
> I'm not saying to use this exact wording but the documentation would make a lot more sense if we wrote it something like this:

Nope, that wouldn't work either because we actually run automated tests against all the config options and variants to ensure that the formatting of the snippet is idempotent when formatted with the corresponding config. I don't really have much more to offer up that hasn't already said beyond general advice when viewing the docs to not make assumptions about things not explicitly stated nor shown, and always feel free to ask clarifying questions!

As mentioned before, when there's a feeling that showing a formatting _diff_ is helpful, the actual "before" is shown explicitly, e.g.
https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#always, but otherwise don't assume there was some different/specific input which was modified to produce the formatted snippets.

**Comment by rukai** [comment]
What if we did it like this?
![image](https://user-images.githubusercontent.com/5120858/132442790-a5ce843e-f046-4d98-adf3-12976592ea66.png)

This way we improve both the idempotentency test and the documentation.

I am willing to provide a PR if its just documentation changes and we find something we both agree on.
